src/pymatgen/io/aims/inputs.py

```diff
@@ -795,25 +768,24 @@ def from_element_and_basis_name(
             f"Can't find the species' defaults file for {element} in {basis} basis set. Paths tried: {paths_to_try}"
         )
 
-    def __str__(self):
+    def __str__(self) -> str:
         """String representation of the species' defaults file"""
         return re.sub(r"^ *species +\w+", f"  species        {self.label}", self.data, flags=re.MULTILINE)
 
     @property
     def element(self) -> str:
-        match = re.search(r"^ *species +(\w+)", self.data, flags=re.MULTILINE)
-        if match is None:
-            raise ValueError("Can't find element in species' defaults file")
-        return match.group(1)
+        if match := re.search(r"^ *species +(\w+)", self.data, flags=re.MULTILINE):
+            return match[1]
+        raise ValueError("Can't find element in species' defaults file")
 
     def as_dict(self) -> dict[str, Any]:
         """Dictionary representation of the species' defaults file."""
         return {"label": self.label, "data": self.data, "@module": type(self).__module__, "@class": type(self).__name__}
 
     @classmethod
-    def from_dict(cls, dct: dict[str, Any]) -> AimsSpeciesFile:
+    def from_dict(cls, dct: dict[str, Any]) -> Self:
         """Deserialization of the AimsSpeciesFile object"""
-        return AimsSpeciesFile(data=dct["data"], label=dct["label"])
+        return cls(**dct)
```

Comment:
this method will now fail if `dct` contains additional keys besides `data` and `label` where it wouldn't before but could be a good safety mechanism unless that leeway was intentional?

Reply:
I think this would be okay. I added the leeway there in case something changed in the future with the species.